torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2527,8 +2527,15 @@ def _sharded_pre_load_state_dict_hook(
             param = state_dict.pop(fqn)
 
             # All-gather the param (ShardedTensor)
-            shards = param.local_shards()
-            local_tensor = cast(torch.Tensor, shards[0].tensor).flatten()
+            param, shards = _tf_pre_load_state_dict_transform(param)
```

Comment:
Can we let tf_flattner/_tf_pre_load_state_dict_transform to implement this logic? All the error handling logic is either in `_create_chunk_sharded_tensor` or `_gather_state_dict`. It makes sense to enforce tf_flattener/_tf_pre_load_state_dict_transform to implement the same logic.

Reply:
Discussed offline: we will keep things as is.